### ISSUE TITLE
loss = nan, imbalanced data

### ISSUE THREAD

**Issue by rmanor** [BVLC/caffe#589]
Hi,

I'm trying to train my data on a net and I always get loss = nan, and also: 
Test score #0: 0.891249
Test score #1: nan

I have binary labels (0,1) and I use Euclidean loss after a sigmoid layer and an accuracy layer after the last sigmoid layer in the validation net.

My data is very unbalanced - 90% of class 0 and 10% of class 1.
Without digging too much in the code, this looks like a case where the batches contain mostly class 0 samples.
Is there some option to balance the batches? or use batch_size = 1 but to sample the two classes with same probability?

If not, can I get a pointer on where I could implement this?
Thank you.
